components/bill/BillDetails.tsx

```diff
@@ -34,7 +34,8 @@ const StyledImage = styled(Image)`
 export const BillDetails = ({ bill }: BillProps) => {
   const { t } = useTranslation("common")
 
-  const isPendingUpgrade = useAuth().claims?.role === "pendingUpgrade"
+  const { user } = useAuth()  // Get the logged-in user (adjust according to your auth logic)
+  const isPendingUpgrade = user?.claims?.role === "pendingUpgrade" // Ensure the role check is based on the authenticated user
```

Comment:
Also, I feel like the comments here are unnecessary - your code seems pretty self-explanatory.